src/tests/Interop/COM/NETClients/VtableGap/ErrorTests.cs

```diff
@@ -0,0 +1,45 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace NetClient
```

Comment:
A test for this already exists: 

https://github.com/dotnet/runtime/blob/04e86c3b8a8925b5ecb8c16a78281b4f412b533e/src/tests/baseservices/typeequivalence/simple/Simple.cs#L137-L150

What is novel here?

Reply:
This test also runs on non-Windows so we validate that the type is crossgened without errors.

Reply:
It's basically a no-op on non-Windows, but it runs (and is crossgenned in crossgen runs).

If we can make that test also run (and no-op) on non-Windows then we could use that test.

Reply:
The written test doesn't really run on non-Windows. See the second branch in `Program.cs`. This is not the appropriate location for this test since it is unrelated to COM. I would instead update the TypeEquivalence tests. Also, there is no need to have any native assets to test this scenario.

Reply:
Yes, the test itself is a no-op, but on non-Windows we're really just testing that crossgen succeeds.

For the type equivalence test, can we update it to run on non-Windows easily? I know you mentioned there were issues with Type Equivalence on + Com Interop off in crosssgen that you hadn't worked through yet last time type equivalence came up.

Reply:
Also, since TypeEquivalence isn't enabled on the non-Windows build of the product, we'd still have to do basically a no-op run on non-Windows.

Reply:
>Yes, the test itself is a no-op, but on non-Windows we're really just testing that crossgen succeeds.

I would instead focus on code that has a chance be eventually running on non-Windows. These tests will never run on a non-Windows platform - way too many assumptions.

>For the type equivalence test, can we update it to run on non-Windows easily?

We should be able to. In fact it would be nice if we enabled Type Equivalence on non-Windows at some point. There were some issues with the runtime build, but I think actual C# compilation should work without issue. All test except the one in question can be disabled on non-Windows. I would start there since this gap logic really is a feature of Type Equivalence.

Reply:
I've updated the type-equivalence test to no-op on non-Windows so it can act as the crossgen test on non-Windows and I've removed the COM test.

Reply:
Since we're getting close to branching, I don't want to enable type equivalence in this PR. I just want to unblock crossgenning crossgen2 and adding a test so we validate that we don't regress this while I wait for my sharedfx sdk PR to go in (which starts crossgenning crossgen2).

Reply:
>I don't want to enable type equivalence in this PR.

Agree. I didn't mean to imply we should do that now, it is a longer term idea.